torchprime/data/dataset.py

```diff
@@ -19,12 +19,33 @@ def make_huggingface_dataset(
   assert isinstance(data, DatasetDict)
   data = data[split]
 
-  column_names = list(data.features)
-  data = data.map(
-    lambda samples: tokenizer(samples["text"]),
-    batched=True,
-    remove_columns=column_names,
-  )
+  def preprocess_batch(batch):
+    inputs, labels, attention_masks = [], [], []
+
+    for q, a in zip(batch["question"], batch["answer"]):
```

Comment:
is this format specific to gsm8k dataset? 
I guess we should have a generic design that is not limited to one dataset